### PR TITLE
add edk2 patches for CVE-2022-36763, CVE-2022-36764, CVE-2022-36765, CVE-2023-45230, CVE-2023-45236, CVE-2023-45232, CVE-2023-45233, CVE-2023-45234, CVE-2023-45235, CVE-2023-45237

### PR DESCRIPTION
**Pull request by rmhsawyer**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Fix edk2 CVEs 

###### Change Log  

- Add upstream [patch](https://github.com/tianocore/edk2/pull/5264) for CVE-2022-36763. Add additional fix-tpm-build-issue-from-CVE-2022-36763.patch to resolve a build issue introduced by CVE-2022-36763.
- Add upstream [patch](https://github.com/tianocore/edk2/commit/aeaee8944f0eaacbf4cdf39279785b9ba4836bb6) for CVE-2022-36765
- Add upstream [patch](https://github.com/tianocore/edk2/commit/f31453e8d6542461d92d835e0b79fec8b039174d) for CVE-2023-45230
- Add upstream [patch](https://github.com/tianocore/edk2/commit/4df0229ef992d4f2721a8508787ebf9dc81fbd6e) for CVE-2023-45232
- Add upstream [patch](https://github.com/tianocore/edk2/commit/1b53515d53d303166b2bbd31e2cc7f16fd0aecd7) for CVE-2023-45234
- Add upstream [patch](https://github.com/tianocore/edk2/commit/fac297724e6cc343430cd0104e55cd7a96d1151e) for CVE-2023-45235
- Add upstream [patch](https://github.com/tianocore/edk2/pull/5582/commits/e7226314bf8e00164e536626a9f813343ca8440e) for CVE-2023-45236
- Add upstream [patch](https://github.com/tianocore/edk2/pull/5582/commits/b2669f14cbc7a8a4d121e2cffa1372aa8d9ae0df) for CVE-2023-45237
- Add nopatch for CVE-2022-36764, CVE-2023-45233 as they fixed in previous patch

###### Does this affect the toolchain?  

NO

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2022-36763
- https://nvd.nist.gov/vuln/detail/CVE-2022-36764
- https://nvd.nist.gov/vuln/detail/CVE-2022-36765
- https://nvd.nist.gov/vuln/detail/CVE-2023-45230
- https://nvd.nist.gov/vuln/detail/CVE-2023-45232
- https://nvd.nist.gov/vuln/detail/CVE-2023-45233
- https://nvd.nist.gov/vuln/detail/CVE-2023-45234
- https://nvd.nist.gov/vuln/detail/CVE-2023-45235
- https://nvd.nist.gov/vuln/detail/CVE-2023-45236
- https://nvd.nist.gov/vuln/detail/CVE-2023-45237

###### Test Methodology

- local package build 
- Buddy Build: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=643700&view=results

### CODE REVIEW

File: SPECS/edk2/edk2.spec [hunk 1]
```diff
@@ -115,6 +115,11 @@ Patch1001: CVE-2023-3817.patch
 Patch1002: CVE-2023-0465.patch
 Patch1003: CVE-2023-2650.patch
 Patch1004: improve-safety-of-DH.patch
+Patch1005: CVE-2022-36763.patch
+Patch1006: CVE-2022-36764.patch
+Patch1007: CVE-2022-36765.patch
+Patch1008: CVE-2023-45230.patch
+Patch1009: CVE-2023-45236.patch
```
Comment (PawelWMS) [inline SPECS/edk2/edk2.spec:122]: Please also update the `%prep` section to apply the patches. It looks like the patches starting from number 1000 and up are reserved for changes in the `CryptoPkg/Library/OpensslLib/openssl` directory. You may need to change the patch number, if these fixes are to be applied directly to the main source folder or tweak the steps in the `%prep` section.

**Review by PawelWMS — CHANGES_REQUESTED** [review]
Need to apply the new .patch files.

File: SPECS/edk2/CVE-2023-45232.nopatch [hunk 2]
```diff
@@ -0,0 +1,2 @@
+CVE already patch in CVE-2023-45230.patch
```
Comment (PawelWMS) [inline SPECS/edk2/CVE-2023-45232.nopatch:1]: For the nopatch files, it is not preferable to drop them and either mention all fixed CVEs in the actually applied patch file (`CVE-2023-45230.patch` in this case) or file a dispute through Astrolabe saying the CVE was fixed in a different patch file.

**Review by tobiasb-ms — APPROVED** [review]

File: SPECS/edk2/CVE-2023-45232.nopatch (hunk 2 above)
Comment (tobiasb-ms) [reply SPECS/edk2/CVE-2023-45232.nopatch:1]: How would mentioning them work? The tooling wouldn't know we had patched the other CVEs right?

Comment (PawelWMS) [reply SPECS/edk2/CVE-2023-45232.nopatch:1]: That's the guidelines I got from the CVE specialists. I'm guessing they might be scanning the patch files. Or at least that's the only thing, that comes to my mind.

File: SPECS/edk2/edk2.spec (hunk 1 above)
Comment (rmhsawyer) [reply SPECS/edk2/edk2.spec]: Good catching. This is resolved

File: SPECS/edk2/CVE-2023-45232.nopatch (hunk 2 above)
Comment (rmhsawyer) [reply SPECS/edk2/CVE-2023-45232.nopatch]: After discussion with CVE team. We are ok to use nopatch method.

**Review by rikenm1 — APPROVED** [review]

**Comment by PawelWMS** [comment]
Failing ptests unrelated to this PR and known issues - not a blocker.

**Review by PawelWMS — APPROVED** [review]

**Review by jslobodzian — APPROVED** [review]

**Pull request merged**